src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java

```diff
@@ -239,26 +252,73 @@ private boolean moveFolder(FileObject srcFile, boolean createNonExistingParents,
                 ArrayList<String> sourceChildrenNames = new ArrayList<>(sourceFileChildren.length);
                 ArrayList<String> destinationChildrenNames = new ArrayList<>(destinationFileChildren.length);
                 for (FileObject child : sourceFileChildren) {
-                    sourceChildrenNames.add(child.getName().getBaseName());
+                    if (child.getType() == FileType.FILE) {
+                        sourceChildrenNames.add(child.getName().getBaseName());
+                    }
                 }
                 for (FileObject child : destinationFileChildren) {
-                    destinationChildrenNames.add(child.getName().getBaseName());
+                    if (child.getType() == FileType.FILE) {
+                        destinationChildrenNames.add(child.getName().getBaseName());
+                    }
                 }
 
                 Collection commonFiles = CollectionUtils.intersection(sourceChildrenNames, destinationChildrenNames);
                 if (!commonFiles.isEmpty()) {
+                    log.error("Folder or one or more sub-directories already exists and overwrite not allowed");
                     return false;
                 }
             }
-        } else {
-            if (createNonExistingParents) {
-                destinationFile.createFolder();
-            } else {
-                log.error("Destination folder does not exist and not configured to create");
+        }
+
+        if (StringUtils.isNotEmpty(filePattern)) {
+            FileObject[] children = srcFile.getChildren();
+            for (FileObject child : children) {
+                boolean result;
+                if (child.getType() == FileType.FILE) {
+                    result = moveFileWithPattern(child, destinationFile, filePattern);
```

Comment:
If the first file returns false and the next file returns true, won't this be true again?

Reply:
Idea is to track if there happens any issue while moving a file. If we make it true for success scenarios, there is no way we can know that something went wrong earlier.